config/initializers/simple_form_bootstrap.rb

```diff
@@ -1,155 +1,439 @@
 # frozen_string_literal: true
+
+# Please do not make direct changes to this file!
+# This generator is maintained by the community around simple_form-bootstrap:
+# https://github.com/rafaelfranca/simple_form-bootstrap
+# All future development, tests, and organization should happen there.
+# Background history: https://github.com/plataformatec/simple_form/issues/1561
+
+# Uncomment this and change the path if necessary to include your own
+# components.
+# See https://github.com/plataformatec/simple_form#custom-components
+# to know more about custom components.
+Dir[Rails.root.join('lib/components/**/*.rb')].each { |f| require f }
+
 # Use this setup block to configure all options available in SimpleForm.
 SimpleForm.setup do |config|
+  # Default class for buttons
+  config.button_class = 'btn'
```

Comment:
Should this be `'btn btn-primary'` or `'btn btn-default'`?

Reply:
only if you want **all** buttons have `btn-primary` style. You could run intro override troubles then.
If you want to use variations like `btn-outline-*` Look at the this [example](https://github.com/m5o/simple_form-bootstrap/blob/d7d279681c27ac7e1fac2485f6db09b4c8be70d1/app/views/examples/verticals/_form.html.erb#L40-L41)

Reply:
My concern though is that the default behavior of `f.button :submit` (i.e. without specifying a class) produces a button which disappears on mouse hover:

![button1](https://user-images.githubusercontent.com/771968/39204681-19b1f2fc-47be-11e8-9a01-ab5c5cb5c40f.png)
![button2](https://user-images.githubusercontent.com/771968/39204682-19e050c0-47be-11e8-88b2-940c4608e8b5.png)
![button3](https://user-images.githubusercontent.com/771968/39204684-1a02c31c-47be-11e8-806e-1c646764feff.png)
![button4](https://user-images.githubusercontent.com/771968/39204687-1a2b16c8-47be-11e8-946e-d7d7badc912d.png)

Checking through the Bootstrap 4 docs, it's not clear to me whether `btn` is meant to be used by itself.  It seems like it's always supposed to be paired with a `btn-*`.

Reply:
Yes, it's supposed to be paired with an additional class. But I understand your concern, with https://github.com/rafaelfranca/simple_form-bootstrap/pull/56/commits/c617d81e05ac856d1fc063763f86b30b082495f1 I've added a style modifier that target elements with only a `.btn` class.

You can use this to avoid specifying a class to your `f.button :submit`

Reply:
That's a good idea.  It might even be nice addition to Bootstrap itself!  For Simple Form though, should it be scoped to `.simple_form [class="btn"]` to prevent contamination?

Also, I'm not sure, but do you know if `.btn[class="btn"]` would have better selector performance than just `[class="btn"]`?  I know that attribute selectors are slower than class selectors, and I know that hierarchical selectors (e.g. `.a .b .c`) are evaluated right-to-left, but I don't know if browsers are smart enough to use the `.btn` part first, then filter on `[class="btn"]` (i.e. evaluate that single selector left-to-right).

Reply:
👍 Try to make a PR to bootstrap.